src/pages/LoginHandler/index.js

```diff
@@ -7,140 +7,141 @@ import {
-}))
-class LoginHandler extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      username: '',
-      password: '',
-      variantVal: 'tertiary',
-      btnColor: 'black',
-      btnDisabled: 'true',
-    };
-  }
+const mapStateToProps = state => {
+  const { auth } = state;
+  return auth;
+};
 
-  componentDidMount = () => {
-    this.disableSubmitBtn();
-  };
+const LoginHandler = props => {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [btnDisabled, setBtnDisabled] = useState(true);
+  const [alertOpen, setAlertOpen] = useState(false);
+  const [alertComponent, setAlertComponent] = useState({
+    message: '',
+    variant: '',
+  });
 
-  setLoggedIn = payload => {
-    const { dispatch } = this.props;
+  const setLoggedIn = payload => {
+    const { dispatch } = props;
     dispatch({
       type: 'auth/loadUser',
       payload,
-    });
-    dispatch(routerRedux.push(`/private`));
-  };
-
-  enableSubmitBtn = () => {
-    this.setState({
-      variantVal: 'primary',
-      btnColor: 'white',
-      btnDisabled: 'false',
-    });
-  };
-
-  disableSubmitBtn = () => {
-    this.setState({
-      variantVal: 'tertiary',
-      btnColor: 'black',
-      btnDisabled: 'true',
+    }).then(() => {
+      dispatch(routerRedux.push(`/private`));
     });
   };
 
-  handleUserNameInputChange = username => {
-    const { password } = this.state;
-    this.setState({
-      username,
+  /* eslint-disable camelcase */
+  const handleLoginSubmit = async () => {
+    const { dispatch } = props;
+    const response = await dispatch({
+      type: 'auth/loginUser',
+      payload: {
+        username,
+        password,
+      },
     });
-    if (password !== '' && username !== '') {
-      this.enableSubmitBtn();
-    } else {
-      this.disableSubmitBtn();
+    const { status, message, auth_token } = response;
+    if (status === 'success') {
+      localStorage.setItem('token', auth_token);
+      setLoggedIn({ username });
     }
-  };
-
-  handlePassWordInputChange = password => {
-    const { username } = this.state;
-    this.setState({
-      password,
+    // shows a toast alert on failed
+    // login attempt only.
+    // a success response populates data
+    // on the global header and should be
+    // enough as a visual cue.
+    setAlertComponent({
+      message,
+      variant: status === 'success' ? 'success' : 'danger',
     });
-    if (username !== '' && password !== '') {
-      this.enableSubmitBtn();
-    } else {
-      this.disableSubmitBtn();
-    }
+    setAlertOpen(true);
   };
 
-  handleLoginSubmit = () => {
-    const { username, password } = this.state;
-    if (username === 'admin' && password === 'admin') {
-      this.setLoggedIn({ username });
-    } else {
-      alert('Wrong username/password pair');
-    }
-  };
+  useEffect(
+    () => {
+      if (username.trim() !== '' && password.trim() !== '') {
```

Comment:
Hey @gurbirkalsi, I converted this component into a function component to leverage `useEffect` hook, which would allow us to validate the form when either username or password changes. if it were a class component, I'd have to call the validate on a `handleLoginSubmit` and `handlePasswordSubmit`.

Reply:
> It seems the Submit button is still enabled here while the password and email field is empty after populating it initially. I believe this functionality was working as intended before, is there a specific reason we converted this class component to a functional component?

Thanks! That was a bug, I forgot to look into.